linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -901,20 +906,48 @@ final class Emitter(config: Emitter.Config, prePrinter: Emitter.PrePrinter) {
       _fullClassChangeTracker.foreach(_.startRun())
     }
 
-    def getCache(version: Version): (DesugaredClassCache, Boolean) = {
+    def getCache(version: Version, hasInstances: Boolean, hasRuntimeTypeInfo: Boolean,
+        hasInstanceTests: Boolean): (DesugaredClassCache, Boolean) = {
```

Comment:
I'm wondering if what is called `FullClassChangeTracker` is not better suited for this purpose? IIUC, it does exactly what we need here for the full class (which is not cached).

Maybe we could rename it to `ModuleChangeTracker` and explicitly document it as "change tracker for uncached trees".

This would also allow to pass it to `needInstanceTests` instead of `classCache`.